src/backend/controllers/purchaseRequests.js

```diff
@@ -0,0 +1,239 @@
+import { db } from "../index.js";
+
+// Get all requests for used book
+export const getRequests = async (req, res) => {
+  const { id: usedBookId } = req.params;
+  const SellerID = req.authorization_id;
+
+  try {
+    const existenceQuery = {
+      text: "SELECT * FROM usedbook WHERE usedbookid = $1",
+      values: [usedBookId],
+    };
+    const existenceResult = await db.query(existenceQuery);
+    if (!existenceResult.rows.length)
+      return res.status(404).json({ error: "Used book not found" });
+    if (SellerID !== existenceResult.rows[0].sellerid && SellerID !== "admin")
+      return res.status(403).json({ error: "You do not own this used book" });
+    const query = {
+      text: "SELECT * FROM purchaserequest WHERE usedbookid = $1",
+      values: [usedBookId],
+    };
+    const result = await db.query(query);
+    const requests = result.rows.map((req) => ({
+      BuyerID: req.buyerid,
+      RequestTimestamp: req.requesttimestamp,
+    }));
+    return res.status(200).json({
+      data: requests,
+    });
+  } catch (error) {
+    return res.status(500).json({ error });
+  }
+};
+
+// Get the request that a user has made for a used book (if anu)
+export const getRequest = async (req, res) => {
```

Comment:
What is the purpose of this API?

Reply:
To retrieve a single user's request for a used book. E.g., when a user clicks on a used book, you may want to either show the user's request for the used book or disable the request button if the user has already requested the book. This needs to be separated from the endpoint that returns all requests for a used book, because we should only show all requests to the seller.

Reply:
你是說 user 是否對這個 used book 發過 purchase request ？

Reply:
yes